src/reducers/sources.js

```diff
@@ -136,12 +139,40 @@ function _updateText(state, action : any) : Record<SourcesState> {
   }));
 }
 
-function removeSourceFromTabList(tabs, id) {
-  return tabs.filter(tab => tab.get("id") != id);
+function addToLocalStorage(url) {
+  if (typeof localStorage !== "undefined") {
+    let tabs = JSON.parse(localStorage.getItem("tabs"));
+    if (!tabs) {
+      tabs = [url];
+    } else if (!tabs.includes(url)) {
+      tabs.push(url);
+    }
+    localStorage.setItem("tabs", JSON.stringify(tabs));
+  }
```

Comment:
I'm not sure, but since `tabs ` is an array of urls, I think we need to initialize the array if `tabs` is not defined and push the url to the array when there's already element(s) inside the array. The array manipulation also applies to removing a url from the array.